src/sinks/file/streaming_sink/compat.rs

```diff
@@ -0,0 +1,83 @@
+use super::StreamingSink;
```

Comment:
What is the reasoning for moving this out of being on the streamingsink type?

Reply:
The rationale is this code is more of a compatibility layer to the pre-async topology. This is partially outlined in the comments to the units inside the mod already. The idea is we can support this pull-based operation mode as a first-class at the topology layer, and it then we will be able to simply remove this mod. The `StreamingSink` type will remain though. For this reason, I think it's a good idea to separate them from the start by design.

If you are asking in regards to why use separate files - I just feel like it's cleaner to have multiple smaller files rather than a big one, especially if they have different concerns, like in this case.